subsys/bluetooth/audio/bap_scan_delegator.c

```diff
@@ -349,6 +362,73 @@ static struct bass_recv_state_internal *bass_lookup_addr(const bt_addr_le_t *add
 	return NULL;
 }
 
+#if defined(CONFIG_BT_BAP_SCAN_DELEGATOR_OVERWRITE_OLDEST)
+static struct bass_recv_state_internal *get_oldest_inactive_state(void)
+{
+	/* Free the oldest updated receive state that is not currently being synced */
+
+	struct bass_recv_state_internal *oldest_inactive_state = NULL;
+
+	for (size_t i = 0U; i < ARRAY_SIZE(scan_delegator.recv_states); i++) {
+		struct bass_recv_state_internal *current_state =
+					&scan_delegator.recv_states[i];
+
+		struct bt_bap_scan_delegator_recv_state *state = &current_state->state;
+
+		bool bis_synced = false;
+
+		if (IS_ENABLED(CONFIG_BT_BAP_BROADCAST_SINK)) {
+			/* First check that if the receive state is associated with a
+			 * broadcast sink, that it is not currently attempting to sync
+			 * to a BIG
+			 */
+			struct bt_bap_broadcast_sink *broadcast_sink =
+					broadcast_sink_get_by_pa(current_state->pa_sync);
+
+			if (broadcast_sink != NULL && broadcast_sink->big != NULL) {
+				/* BIG is being synced */
+				LOG_DBG("Receive state at index %zu has an associated big. "
+						"Unable to free this", i);
+				continue;
+			}
+		}
+
+		/* Check the bis_sync for each subgroup. If it is 0 for all subgroups
+		 * that receive state is not in sync with a broadcast source
+		 */
```

Comment:
Since we are checking the `big` right above, can't we skip the checks for `bis_sync` here? A subgroup cannot be synced if the sink itself isn't synced (which it will never be if `sink->big == NULL`.